### PR TITLE
[SPARK-47577][SPARK-47579] Correct misleading usage of log key TASK_ID

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Correct misleading usage of log key TASK_ID from https://github.com/apache/spark/pull/45834 and https://github.com/apache/spark/pull/46739

### Why are the changes needed?

Provide more accurate log keys in the structure logging

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Existing GA tests
### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala [hunk 1]
```diff
@@ -989,7 +989,7 @@ private[spark] class TaskSetManager(
         if (ef.className == classOf[TaskOutputFileAlreadyExistException].getName) {
           // If we can not write to output file in the task, there's no point in trying to
           // re-execute it.
-          logError(log"Task ${MDC(TASK_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
+          logError(log"Task ${MDC(TASK_INFO_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
             log"(TID ${MDC(TID, tid)}) can not write to output file: " +
```
Comment (pan3793) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:993]: what's the different between TID and TASK_ID?

Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:993]: task info id is `def id: String = s"$index.$attemptNumber"`

Comment (pan3793) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:993]: I'm actually pointing #L993

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala]: Good point, I unified TID and TASK_ID

File: core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala [hunk 2]
```diff
@@ -989,8 +989,8 @@ private[spark] class TaskSetManager(
         if (ef.className == classOf[TaskOutputFileAlreadyExistException].getName) {
           // If we can not write to output file in the task, there's no point in trying to
           // re-execute it.
-          logError(log"Task ${MDC(TASK_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
-            log"(TID ${MDC(TID, tid)}) can not write to output file: " +
+          logError(log"Task ${MDC(TASK_INFO_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
```
Comment (pan3793) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:992]: the `taskSet.id` is `s"$stageId.$stageAttemptId"`

Comment (pan3793) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:992]: for cases like TASK_INFO_ID and TASK_SET_ID, I wondering if we can just expose

- TASK_INDEX
- TASK_ATTEMP_NUM
- STAGE_ID
- STAGE_ATTEMP_ID

to the MDC?

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala]: Good point, updated.

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala [hunk 3]
```diff
@@ -2019,7 +2019,7 @@ private[spark] class DAGScheduler(
           val ignoreStageFailure = ignoreDecommissionFetchFailure &&
             isExecutorDecommissioningOrDecommissioned(taskScheduler, bmAddress)
           if (ignoreStageFailure) {
-            logInfo(log"Ignoring fetch failure from ${MDC(TASK_ID, task)} of " +
+            logInfo(log"Ignoring fetch failure from ${MDC(TASK_NAME, task)} of " +
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala:2022]: Do we really want to call this `TASK_NAME`?
From the code below, it seems inconsistent:
https://github.com/apache/spark/blob/33a9c5d7f478a476fe9882ad8fe101fd60756a98/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L839
https://github.com/apache/spark/blob/33a9c5d7f478a476fe9882ad8fe101fd60756a98/core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala#L583-L587

File: core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala [hunk 4]
```diff
@@ -989,9 +989,11 @@ private[spark] class TaskSetManager(
         if (ef.className == classOf[TaskOutputFileAlreadyExistException].getName) {
           // If we can not write to output file in the task, there's no point in trying to
           // re-execute it.
-          logError(log"Task ${MDC(TASK_ID, info.id)} in stage ${MDC(STAGE_ID, taskSet.id)} " +
-            log"(TID ${MDC(TID, tid)}) can not write to output file: " +
-            log"${MDC(ERROR, ef.description)}; not retrying")
+          logError(
+            log"Task ${MDC(TASK_INDEX, info.index)}.${MDC(TASK_ATTEMPT_ID, info.attemptNumber)} " +
+              log"in stage ${MDC(STAGE_ID, taskSet.stageId)}." +
```
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:994]: Can we unify the writing style of `taskSet.logid`?
image
the current writing style more consistent with traditional styles.

File: core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala (hunk 3 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala:2022]: well, I can't find a better name here..

File: core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala (hunk 4 above)
Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala:994]: This one only shows once...I was thinking about not having a new method

File: core/src/main/scala/org/apache/spark/scheduler/TaskSet.scala [hunk 5]
```diff
@@ -34,4 +37,12 @@ private[spark] class TaskSet(
   val id: String = s"$stageId.$stageAttemptId"
 
   override def toString: String = "TaskSet " + id
+
+  // Identifier used in the structured logging framework.
+  lazy val logId: MessageWithContext = {
```
Comment (cloud-fan) [inline core/src/main/scala/org/apache/spark/scheduler/TaskSet.scala:42]: So we intentionally not treat it as a log parameter?

Comment (gengliangwang) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSet.scala:42]: what do you mean by "log parameter"?

Comment (cloud-fan) [reply core/src/main/scala/org/apache/spark/scheduler/TaskSet.scala:42]: nvm, it's a `MessageWithContext`

**Review by cloud-fan — APPROVED** [review]

**Comment by gengliangwang** [comment]
@pan3793 @panbingkun @cloud-fan Thanks for the review.
Merging to master

**Pull request closed without merging**